src/main/kotlin/nl/avisi/structurizr/site/generatr/site/model/DecisionTabViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package nl.avisi.structurizr.site.generatr.site.model
+
+import com.structurizr.model.SoftwareSystem
+import nl.avisi.structurizr.site.generatr.hasDecisions
+
+data class DecisionTabViewModel(val pageViewModel: SoftwareSystemPageViewModel, val title: String, val url: String) {
+    val link = LinkViewModel(pageViewModel, title, url, true)
+}
+
+fun SoftwareSystemPageViewModel.createDecisionsTabViewModel(softwareSystem: SoftwareSystem, tab: SoftwareSystemPageViewModel.Tab): List<DecisionTabViewModel> {
+    val tabs = buildList<DecisionTabViewModel> {
+        if (softwareSystem.hasDecisions()) {
+            add(DecisionTabViewModel(
+                    this@createDecisionsTabViewModel,
+                    "System",
+                    SoftwareSystemPageViewModel.url(softwareSystem, tab)
+            ))
+        }
+        softwareSystem
+                .containers
```

Comment:
Right now, a decision tab is added for every container, regardless of whether or not it has any decisions. You'll need to add a `.filter` to make sure that decision tabs are only generated for containers that have decisions:

```suggestion
                .containers
                .filter { it.hasDecisions() }
```